visualization/app/codeCharta/util/algorithm/treeMapLayout/treeMapHelper.ts

```diff
@@ -203,13 +204,91 @@ export function getBuildingColor(node: CodeMapNode, { appSettings, dynamicSettin
 		return mapColors.flat
 	}
 
-	if (metricValue < dynamicSettings.colorRange.from) {
+	const positiveColorRGB = ColorConverter.convertHexToColorObject(mapColors.positive)
+	const neutralColorRGB = ColorConverter.convertHexToColorObject(mapColors.neutral)
+	const negativeColorRGB = ColorConverter.convertHexToColorObject(mapColors.negative)
+
+	const { colorRange, colorMode } = dynamicSettings
+
+	switch (colorMode) {
+		case ColorMode.trueGradient: {
+			return calculateTrueGradient(colorRange, metricValue, positiveColorRGB, neutralColorRGB, negativeColorRGB)
+		}
+
+		case ColorMode.weightedGradient: {
+			return calculateWeightedGradient(mapColors, colorRange, metricValue, positiveColorRGB, neutralColorRGB, negativeColorRGB)
+		}
+
+		case ColorMode.absolute:
+			if (metricValue < colorRange.from) {
+				return mapColors.positive
+			}
+			if (metricValue > colorRange.to) {
+				return mapColors.negative
+			}
+			return mapColors.neutral
+	}
+}
+
+function calculateTrueGradient(
+	colorRange: { from: any; to: any; max: number },
+	metricValue: number,
+	positiveColorRGB: Color,
+	neutralColorRGB: Color,
+	negativeColorRGB: Color
+) {
+	const middle = (colorRange.from + colorRange.to) / 2
+
+	if (metricValue <= middle) {
+		const factor = metricValue / middle
+		return ColorConverter.convertColorToHex(new Color().lerpColors(positiveColorRGB, neutralColorRGB, factor))
+	}
+
+	if (metricValue <= colorRange.max && metricValue > middle) {
+		const factor = (metricValue - middle) / (colorRange.max - middle)
+		return ColorConverter.convertColorToHex(new Color().lerpColors(neutralColorRGB, negativeColorRGB, factor))
+	}
```

Comment:
Would it be possible that nothing will be returned, because no if-statement is executed?